odoo/models.py

```diff
@@ -4506,6 +4518,82 @@ def _parent_store_update(self):
         self.env.cache.update(records, self._fields['parent_path'], updated.values())
         records.modified(['parent_path'])
 
+    def _identify_existing_records(self, data_list, search_fields):
+        ''' Find or create an External ID (xml_id) for each of the unique records
+            having matching data in search_fields
+
+            :param data_list: list of dicts with keys `xml_id` and record values
+            :param search_fields: list of technical field names
+
+            :return: updated data_list containing the new/found xml_id
+
+            :raises ValidationError: when multiple records match the search criteria
+        '''
+        # check if search_fields exist in the models and filter out everything else preventing error and SQL injection
+        search_fields = [f for f in search_fields if f in self._fields.keys()]
+        if not search_fields:
+            return data_list
+
+        key_fields = ', '.join(search_fields)
+        prefixed_key_fields = ', '.join([f'r.{field}' for field in search_fields])
+        key_data = [tuple(item['values'][field] for field in search_fields) for item in data_list]
+
+        sql = f"""
+            CREATE TEMP TABLE temp_{self._table}(
```

Comment:
are you nuts?

Reply:
it seems so, yes :) but I changed it now